appeals/web/src/client/components/file-uploader/_client-actions.js

```diff
@@ -258,25 +254,36 @@ const clientActions = (container) => {
 			return;
 		}
 
+		stagedFiles.errors.length = 0;
+
+		if (allowSingleFileOnly() && (fileList.length > 1 || stagedFiles.files?.length)) {
+			await showErrors(
+				{
+					message: 'FILE_SPECIFIC_ERRORS',
+					details: [
+						{
+							message: 'SINGLE_FILE_ONLY',
+							guid: '',
+							name: '',
+							formId: container.dataset?.formId || ''
+						}
+					]
+				},
+				container
+			);
+			updateStagedFilesUI(stagedFiles);
+			return;
+		}
+
 		showProgressMessage(container);
 
-		let addedFiles = Array.from(fileList).map((file) => ({
+		let addedFiles = Array.from(fileList || []).map((file) => ({
```

Comment:
I think this can be `const` now that it isn't reassigned.
```suggestion
		const addedFiles = Array.from(fileList || []).map((file) => ({
```